content: update logos to new branding

djacu opened the pull request:
Replaces the logo files with the updated version.

Summary of files removed or updated:

```
./core/public/favicon.svg                               removed (we are not using this)
./core/public/favicon.png                               removed (we are not using this)

./core/src/assets/image/nixos-logo-text.svg             removed (using combined logo)
./core/src/assets/image/nixos-logo-notext.svg           updated
./core/src/assets/image/nix-snowflake-rainbow.svg       updated
./core/src/assets/image/flake-blue.svg                  removed (no usage)
./core/src/assets/image/nixos-foundation-logo.svg       TODO no update for now
./core/src/assets/image/flake-white.svg                 updated

./core/public/images/teams/nixpkgs-architecture.svg     removed (no team listed)
./core/public/images/teams/ngi.svg                      updated
./core/public/images/teams/documentation.svg            updated
./core/public/images/teams/formatting.svg               updated

./core/public/images/logos/nixos_flake.svg              removed

./core/public/logo/nixos-hires.png                      removed (no usage)
./core/public/logo/nixos-logo-only-hires.png            updated

./core/src/assets/logo/nixos-hex.svg                    removed (no usage)
./core/src/assets/logo/nix-wiki.png                     removed (no usage)
./core/src/assets/logo/nixos-hires-print.png            removed (no usage)
./core/src/assets/logo/nixos-hires.png                  removed (no usage)
./core/src/assets/logo/nixos-logo-only-hires.png        removed (no usage)
./core/src/assets/logo/nixos-logo-only-hires-print.png  removed (no usage)
./core/src/assets/logo/nixos-lores-no-alpha.png         removed (no usage)
./core/src/assets/logo/nixos-lores.png                  legacy (no touch - used for old blog post)
```

We now have assets for the NixOS logo and logomark with default, rainbow, black, and white color variants. All of these have no clearspace so it must be added with styling.

thilobillerbeck reviewed (changes requested):
Looks alright

@djacu do we really want to use the full width logo instead of the flake for the favicon?

djacu commented:
@thilobillerbeck 

I'm confused. I replaced the favicon with the flake (logomark). That's what we wanted right?

![image](https://github.com/user-attachments/assets/9a0f81c6-0a35-4b10-b72e-75336f5971de)

djacu commented:
Oh but it is showing up as the full logo.... the heck

djacu commented:
Oh I see, `getNixLogoUrlUniversal` was previously only returning just the logomark. And that was combined with the "Nix" SVG previously in the header. I understand now. Will fix.

djacu commented:
@thilobillerbeck do we need to keep the favicon.svg file around or can it be removed?

thilobillerbeck commented:
Ahh, yit can ve removed, though we need a Flake only SVG for the favicon generation. There should be some hints in the Astro config where they live.

djacu commented:
Alright should be working as intended now. I added additional white and black variants of the logo and updated the universal NixOS logo getter function.

thilobillerbeck reviewed (approved):
awesome! <3
